File: src/test/java/com/tartner/domain/password/PasswordQueryModelTest.java
package com.tartner.domain.password;

import com.tartner.dancehours.querymodel.jpa.AggregatePasswordsEntity;
import org.junit.Before;
import org.junit.Test;

import java.security.NoSuchAlgorithmException;
import java.util.UUID;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class PasswordQueryModelTest {

    public static final String TestPassword = "P@ssw0rd";

    private static final int SaltLength = 32;

    private PasswordService passwordService;
    private PasswordQueryModel queryModel;
    private AggregatePasswordRepository repository;

    @Before
    public void setUp() throws NoSuchAlgorithmException {
        passwordService = new PasswordService();
        repository = mock( AggregatePasswordRepository.class );
        queryModel = new PasswordQueryModel( passwordService, repository );
    }

    @Test
    public void testPasswordsMatch() throws Exception {
        UUID id = UUID.randomUUID();

        TestPasswordHolder holder = TestPasswordHolder.CreateDefaultTest();

        AggregatePasswordsEntity password = new AggregatePasswordsEntity();
        password.setAggregateId( id );
        password.setPasswordHash( holder.getPasswordHash() );
        password.setSalt( holder.getSalt() );

        when( repository.findOne( id ) ).thenReturn( password );

        final boolean passwordsMatch =
            queryModel.passwordsMatch( id, TestPassword );
        assertThat( passwordsMatch, is( true ) );
    }
}

File: src/test/java/com/tartner/domain/password/PasswordQueryModelTest.kt
package com.tartner.domain.password

import com.tartner.dancehours.querymodel.jpa.AggregatePasswordsEntity
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock
import java.security.NoSuchAlgorithmException
import java.util.*

public class PasswordQueryModelTest {

    private var passwordService: PasswordService? = null
    private var queryModel: PasswordQueryModel? = null
    private var repository: AggregatePasswordRepository? = null

    @Before
    @throws(NoSuchAlgorithmException::class)
    public fun setUp() {
        passwordService = PasswordService()
        repository = mock(javaClass<AggregatePasswordRepository>())
        queryModel = PasswordQueryModel(passwordService!!, repository!!)
    }

    @Test @throws(Exception::class)
    public fun testPasswordsMatch() {
        val id = UUID.randomUUID()

        val holder = TestPasswordHolder.CreateDefaultTest()

        val password = AggregatePasswordsEntity()
        password.aggregateId = id
        password.passwordHash = holder.passwordHash
        password.salt = holder.salt

        `when`(repository!!.findOne(id)).thenReturn(password)

        val passwordsMatch = queryModel!!.passwordsMatch(id, TestPassword)
        assertThat(passwordsMatch, `is`(true))
    }

    companion object {
        public val TestPassword: String = "P@ssw0rd"
        private val SaltLength = 32
    }
}
